autofit/graphical/messages/abstract.py

```diff
@@ -73,45 +79,51 @@ def shape(self) -> Tuple[int, ...]:
         return self._broadcast.shape
 
     @property
-    def ndim(self) -> int: 
+    def size(self) -> int:
+        return self._broadcast.size
+
+    @property
+    def ndim(self) -> int:
         return self._broadcast.ndim
 
     @classmethod
     def from_natural_parameters(
-            cls, 
-            parameters: Tuple[np.ndarray, ...], 
+            cls,
+            natural_parameters: np.ndarray,
             **kwargs
     ) -> "AbstractMessage":
-        args = cls.invert_natural_parameters(parameters)
+        cls = cls._projection_class or cls
+        args = cls.invert_natural_parameters(natural_parameters)
         return cls(*args, **kwargs)
 
     @classmethod
     @abstractmethod
     def invert_sufficient_statistics(cls, sufficient_statistics: np.ndarray
-    ) -> Tuple[np.ndarray, ...]:
+                                     ) -> np.ndarray:
         pass
 
     @classmethod
     def from_sufficient_statistics(cls, suff_stats: np.ndarray, **kwargs
-    ) -> "AbstractMessage":
+                                   ) -> "AbstractMessage":
         natural_params = cls.invert_sufficient_statistics(suff_stats)
+        cls = cls._projection_class or cls
         return cls.from_natural_parameters(natural_params, **kwargs)
 
     def sum_natural_parameters(self, *dists: "AbstractMessage"
-    ) -> "AbstractMessage":
+                               ) -> "AbstractMessage":
```

Comment:
Something looks a bit weird with the formatting here

Reply:
I think that's an Autopep8 casualty

Reply:
Ha ok. I guess I'd put self, dists and the closing bracket each on a new line. AFAIK that's pep compliant

Reply:
This is pep8 compliant, but I agree it's non-standard for Autofit (and most other codes). 

I'll put `self, dists` onto a new line to make it more inline with the rest of the codebase.